modules/system/effect-wfrp4e.js

```diff
@@ -585,9 +585,13 @@ export default class EffectWfrp4e extends ActiveEffect
         {
             return test;
         }
+        else if (test?.data)
+        {
+            return TestWFRP.recreate(test.data);
+        }
```

Comment:
This is risky and I don't think it will work. Scripts should expect `sourceTest` to be serialized test data. If you know of any scripts that don't assume that (like trying to use getters) let me know.

Reply:
Why this is risky? I noticed one effect that was inconsistent. When run without socket it was getting testobject and worked fine, and on socket it was only getting testdata. I cannot be 100% if this wasn't custom effect, but for sure inconsistency was there

Reply:
It's risky because if tests are created during world load they may fail due to referencing documents that aren't ready yet. 

I've changed all instances of `this.effect.sourceTest.data.result` to `this.effect.sourceTest.result` if that's what you mean by inconsistency.